adala/utils/llm_utils.py

```diff
@@ -139,13 +175,19 @@ def run_instructor_with_messages(
         temperature: Temperature for sampling
         seed: Integer seed to reduce nondeterminism
         retries: Retry policy to use
+        ensure_messages_fit_in_context_window: Whether to ensure the messages fit in the context window (setting it to True will slow down the function)
         **kwargs: Additional arguments to pass to the completion call
 
     Returns:
         Dict containing the parsed response and usage information
     """
     try:
-        # returns a pydantic model and completion info
+        prompt_token_count = None
+        if ensure_messages_fit_in_context_window:
+            messages, prompt_token_count = _ensure_messages_fit_in_context_window(
+                messages, model
+            )
```

Comment:
@niklub here is where `self.canonical_model_provider_string` should be used. here is the implementation: https://github.com/HumanSignal/Adala/blob/master/adala/utils/model_info_utils.py#L48

The regular self.model is the "endpoint" model name that the model provider understands. The canonical string is the "canonical" model name that litellm itself understands, where the model is listed with its supported parameters, cost per token, context window, etc in a big lookup table.

Reply:
They are often different :upside_down_face: